daisy/disk.go

```diff
@@ -237,3 +256,29 @@ func (dr *diskRegistry) regDetachAll(iName string, s *Step) DError {
 	}
 	return errs
 }
+
+// Merges two slices of features and returns a new slice instance.
+// Duplicates are removed.
+func combineGuestOSFeatures(features1 []*compute.GuestOsFeature,
+	features2 ...string) []*compute.GuestOsFeature {
+
+	featureSet := map[string]bool{}
+	for _, feature := range features2 {
+		featureSet[feature] = true
+	}
+	for _, feature := range features1 {
+		featureSet[feature.Type] = true
+	}
+	ret := make([]*compute.GuestOsFeature, 0)
+	for feature := range featureSet {
+		ret = append(ret, &compute.GuestOsFeature{
+			Type: feature,
+		})
+	}
+	// Sort elements by type, lexically. This ensures
+	// stability of output ordering for tests.
+	sort.Slice(ret, func(i, j int) bool {
```

Comment:
I still I don't agree we should be doing anything just to make tests happy (like this sorting) as it's boilerplate code that should belong in the test (client code of this function doesn't require it so it shouldn't be there). If you copy-pasted this code in the tests before asserting, tests will succeed as is now and we won't do unnecessary things in real code. It will also save future developers of this code trying to understand why are we sorting the list.